src/docfx/lib/markdown/ResolveXref.cs

```diff
@@ -48,6 +50,11 @@ public static MarkdownPipelineBuilder UseResolveXref(this MarkdownPipelineBuilde
                          var displayPropertyValue = xrefSpec.GetXrefPropertyValue(queries?["displayProperty"]);
                          string display = !string.IsNullOrEmpty(displayPropertyValue) ? displayPropertyValue : (!string.IsNullOrEmpty(name) ? name : uid);
                          var href = !string.IsNullOrEmpty(moniker) ? $"{xrefSpec.Href}?view={moniker}" : xrefSpec.Href;
+
+                         // append fragment if any
+                         if (!string.IsNullOrEmpty(fragment))
+                             href += fragment;
```

Comment:
>href [](start = 29, length = 4)

href may also contains fragment

Reply:
There are other places where we need this merge functionality

Reply:
Implemented with unit tests